client/dashboard/domains/domain-overview/index.tsx

```diff
@@ -0,0 +1,65 @@
+import { useQuery, useSuspenseQuery } from '@tanstack/react-query';
+import { Button } from '@wordpress/components';
+import { __, sprintf } from '@wordpress/i18n';
+import { useMemo } from 'react';
+import { useLocale } from '../../app/locale';
+import { domainQuery } from '../../app/queries/domain';
+import { sitePurchaseQuery } from '../../app/queries/site-purchases';
+import { domainRoute } from '../../app/router';
+import { PageHeader } from '../../components/page-header';
+import PageLayout from '../../components/page-layout';
+import { formatDate } from '../../utils/datetime';
+import { getDomainRenewalUrl, isDomainRenewable } from '../../utils/domain';
+
+export default function DomainOverview() {
+	const locale = useLocale();
+	const { domainName } = domainRoute.useParams();
+	const { data: domain } = useSuspenseQuery( domainQuery( domainName ) );
+	const { data: purchase } = useQuery(
+		sitePurchaseQuery( domain.blog_id, parseInt( domain.subscription_id, 10 ) )
+	);
+
+	const wrappableDomainName = useMemo( () => {
+		const [ domainName, ...tlds ] = domain.domain.split( '.' );
+		const tld = tlds.join( '.' );
+
+		return (
+			<span style={ { wordBreak: 'break-word', hyphens: 'none' } }>
+				{ domainName }
+				<span style={ { whiteSpace: 'nowrap' } }>.{ tld }</span>
+			</span>
+		);
+	}, [ domain.domain ] );
+
+	return (
+		<PageLayout
+			size="small"
+			header={
+				<PageHeader
+					title={ wrappableDomainName }
+					// translators: date is the date the domain was registered.
+					description={ sprintf( __( 'Registered on %(date)s' ), {
+						date: formatDate( new Date( domain.registration_date ), locale, { dateStyle: 'long' } ),
+					} ) }
+					actions={
+						isDomainRenewable( domain ) &&
```

Comment:
@arthur791004 @fushar @sirbrillig, why do we need this `isDomainRenewable` function if `purchase` has an `can_explicitly_renew` property? Are these two different things? Can I even use this function, which is [borrowed from the dataview](https://github.com/Automattic/wp-calypso/blob/trunk/client/dashboard/domains/dataviews/actions.tsx#L71)?

Reply:
I just followed the existing logic 🙈